src/AssetRegistry/ATokenPricingModule.sol

```diff
@@ -58,30 +66,108 @@ contract ATokenPricingModule is PricingModule {
      * assets are no longer updatable.
      * @dev Assets can't have more than 18 decimals.
      */
-    function setAssetInformation(
-        address assetAddress,
-        uint16[] calldata assetCollateralFactors,
-        uint16[] calldata assetLiquidationThresholds
-    ) external onlyOwner {
-        address underlyingAddress = IAToken(assetAddress).UNDERLYING_ASSET_ADDRESS();
-        (uint64 assetUnit, address underlyingAssetAddress, address[] memory underlyingAssetOracleAddresses) =
-        IPricingModule(IMainRegistry(mainRegistry).assetToPricingModule(underlyingAddress)).getAssetInformation(
-            underlyingAddress
-        );
+    function setAssetInformation(AssetInformation memory assetInformation) external onlyOwner {
```

Comment:
Might be cleaner to input variables separately and not in struct, and just fetch the parameters such as underlyingAssetUnit instead of checking it is equal.

Might be a different task.

Reply:
#157